File: MPP-lab-project/src/Repository/DBRepository/AdminDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using log4net;
using log4net.Config;
using log4net.Util;
using MPP_lab_project.Domain;
using MPP_lab_project.Repository.Interfaces;
using MPP_lab_project.Utils;

namespace MPP_lab_project.Repository.DBRepository;

public class AdminDBRepository : IAdminRepository
{
    private static readonly ILog Logger = LogManager.GetLogger("AdminDBRepository");
    IDictionary<String, String> properties;

    public AdminDBRepository(IDictionary<String, String> properties)
    {
        Logger.InfoFormat("Initialising AdminDBRepository...");
        this.properties = properties;
    }

    public int Add(Admin elem)
    {
        throw new System.NotImplementedException();
    }

    public void Delete(Admin elem)
    {
        throw new System.NotImplementedException();
    }

    public void Update(Admin elem, int id)
    {
        throw new System.NotImplementedException();
    }

    public Admin FindById(int id)
    {
        throw new System.NotImplementedException();
    }

    public Admin FindByUsernameAndPassword(String username, String password)
    {
        Logger.InfoFormat("FindByUsernameAndPassword(username = {0}, password = {1})", username, password);
        Admin admin = null;

        IDbConnection connection = DbUtils.GetConnection(properties);
        using (IDbCommand comm = connection.CreateCommand())
        {
            comm.CommandText = "select id from Admins where username = @username and password = @password;";
        
            IDbDataParameter paramUsername = comm.CreateParameter();
            paramUsername.ParameterName = "@username";
            paramUsername.Value = username;
            comm.Parameters.Add(paramUsername);
        
            IDbDataParameter paramPassword = comm.CreateParameter();
            paramPassword.ParameterName = "@password";
            paramPassword.Value = password;
            comm.Parameters.Add(paramPassword);

            using (IDataReader dataReader = comm.ExecuteReader())
            {
                if (dataReader.Read())
                {
                    Int32 id = dataReader.GetInt32(0);
                    admin = new Admin(id, username, password);
                }
            }
        }
        Logger.InfoFormat("Result: admin = {0}", admin);
        
        return admin;
    }
}

File: swimming-contest-software/Server/src/main/java/swimmingApp/repository/dbRepository/AdminDBRepository.java
package swimmingApp.repository.dbRepository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import swimmingApp.domain.entities.Admin;
import swimmingApp.repository.interfaces.AdminRepository;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class AdminDBRepository implements AdminRepository {

    private final JdbcUtils jdbcUtils;
    private static final Logger logger = LogManager.getLogger();

    public AdminDBRepository(Properties properties) {
        logger.info("Initialising AdminDBRepository with properties {}", properties);
        this.jdbcUtils = new JdbcUtils(properties);
    }

    @Override
    public Integer add(Admin elem) {
        return null;
    }

    @Override
    public void delete(Admin elem) {

    }

    @Override
    public void update(Admin elem, Integer id) {

    }

    @Override
    public Admin findById(Integer id) {

        return null;
    }

    @Override
    public Admin findByUsernameAndPassword(String username, String password) {
        logger.traceEntry("findByUsernameAndPassword(username = {}, password {})", username, password);

        Admin admin = null;

        Connection connection = jdbcUtils.getConnection();
        try (PreparedStatement preparedStatement = connection.prepareStatement(
                "select id from main.Admins where username = ? and password = ?;"
        )) {
            preparedStatement.setString(1, username);
            preparedStatement.setString(2, password);
            ResultSet resultSet = preparedStatement.executeQuery();
            if (resultSet.next()) {
                Integer id = resultSet.getInt("id");
                admin = new Admin(id, username, password);
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }

        return logger.traceExit("Result: admin = {}", admin);
    }
}
